app/frontend/actions.ts

```diff
@@ -1347,54 +1333,32 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
         throw NamedError('PoolRegNoHwWallet')
       }
 
+      const {plan, ttl, validityIntervalStart} = state.poolRegTransactionSummary
+
       const txAux = await wallet.getAccount(state.sourceAccountIndex).prepareTxAux(
-        state.poolCertTxVars.plan, // @ts-ignore (Fix byron-shelley formats later)
-        parseInt(state.poolCertTxVars.ttl, 10)
+        plan,
+        // TODO: we should do parsing somewhere else
+        parseCliTtl(ttl),
```

Comment:
I'd expect to do that at the moment of creation of the poolRegTransactionSummary and it can be casted to some branded type like ValidTtl which internal functions then can take for granted to be actually valid

Reply:
Fixed, parsing ttl and validity interval start somewhere else, I didn't create the branded type tho.